application/src/main/java/org/eclipse/hono/application/ApplicationConfig.java

```diff
@@ -29,16 +30,31 @@
 import io.vertx.core.Vertx;
 import io.vertx.core.VertxOptions;
 import io.vertx.core.dns.AddressResolverOptions;
+import io.vertx.core.metrics.MetricsOptions;
 
 /**
  * Spring bean definitions required by the Hono application.
  */
 @Configuration
 public class ApplicationConfig {
 
+    private MetricsOptions metricsOptions;
+
+    /**
+     * Adds metrics options to the Vert.x configuration, to write the Vert.x metrics to the configured Hono metrics
+     * server
+     *
+     * @param metricsOptions configured metrics options for Vert.x
+     * @see MetricsConfig
+     */
+    @Autowired(required = false)
+    public ApplicationConfig(final MetricsOptions metricsOptions) {
```

Comment:
this doesn't seem to work when there is no bean of type `MetricsOptions` being created, i.e. because the necessary property has not been set :-(
Seems like we will need to store the `MetricsOptions` in an autowired(required = false) field instead (omitting the constructor altogether).